File: mcp_status_parser.js
#!/usr/bin/env node
const fs = require('fs');
const path = require('path');

const STATUS_FILE = path.resolve(__dirname, 'mcp_status.json');

function loadStatus() {
  if (!fs.existsSync(STATUS_FILE)) {
    console.error(`Status file not found: ${STATUS_FILE}`);
    process.exit(1);
  }
  const text = fs.readFileSync(STATUS_FILE, 'utf8');
  return JSON.parse(text);
}

function formatNumber(value) {
  if (typeof value !== 'number') return String(value);
  if (value >= 1e12) return `${(value / 1e12).toFixed(2)}T`;
  if (value >= 1e9) return `${(value / 1e9).toFixed(2)}B`;
  if (value >= 1e6) return `${(value / 1e6).toFixed(2)}M`;
  if (value >= 1e3) return `${(value / 1e3).toFixed(2)}K`;
  return value.toFixed(2);
}

function main() {
  const status = loadStatus();
  console.log('mcp status parser');
  console.log('==================');
  console.log(`timestamp:   ${new Date(status.ts).toISOString()}`);
  console.log(`target:      ${status.target}`);
  console.log(`plan:        ${status.plan}`);
  console.log(`sec:         ${status.currentSecurity.toFixed(2)}`);
  console.log(`moneyPct:    ${status.moneyPct.toFixed(3)}`);
  console.log(`needWeaken:  ${status.needWeaken}`);
  console.log(`maxWeaken:   ${status.maxWeaken}`);
  console.log(`homeFreeRam: ${status.homeFreeRam.toFixed(2)} GB`);
  console.log(`hacked:      ${formatNumber(status.hacked)}`);
  console.log(`rate:        ${formatNumber(status.rate)}/s`);
  console.log(`avgRate:     ${formatNumber(status.avgRate)}/s`);
  console.log(`total:       ${formatNumber(status.totalHacked)}`);
  console.log(`workers:     ${status.workers.length}`);
  console.log(`candidate:   ${status.candidate}`);
  console.log(`candidateScore: ${formatNumber(status.candidateScore)}`);
  console.log(`expectedIncome: ${formatNumber(status.candidateExpectedIncome)}/s`);
  console.log('');
  console.log('workers:');
  for (const worker of status.workers) {
    const actions = worker.actions && worker.actions.length > 0
      ? worker.actions.map(a => `${a.script}(${a.threads})`).join(', ')
      : 'idle'
    console.log(`  - ${worker.host}: freeRam=${worker.freeRam.toFixed(2)} GB usedRam=${worker.usedRam.toFixed(2)} GB maxRam=${worker.maxRam.toFixed(2)} GB actions=${actions}`);
  }
}

main();

File: mcp_status_parser.py
import json
from pathlib import Path

STATUS_FILE = Path(__file__).resolve().parent / 'mcp_status.json'

def fmt_num(value):
    if not isinstance(value, (int, float)):
        return str(value)
    if value >= 1e12:
        return f'{value/1e12:.2f}T'
    if value >= 1e9:
        return f'{value/1e9:.2f}B'
    if value >= 1e6:
        return f'{value/1e6:.2f}M'
    if value >= 1e3:
        return f'{value/1e3:.2f}K'
    return f'{value:.2f}'


def main():
    if not STATUS_FILE.exists():
        print(f'Status file not found: {STATUS_FILE}')
        return

    status = json.loads(STATUS_FILE.read_text())
    print('mcp status parser')
    print('==================')
    print(f"timestamp:        {status['ts']} ({status['ts'] and __import__('datetime').datetime.utcfromtimestamp(status['ts']/1000).isoformat()+'Z'})")
    print(f"target:           {status.get('target')}")
    print(f"plan:             {status.get('plan')}")
    print(f"sec:              {status.get('currentSecurity', 0):.2f}")
    print(f"moneyPct:         {status.get('moneyPct', 0):.3f}")
    print(f"needWeaken:       {status.get('needWeaken')}")
    print(f"maxWeaken:        {status.get('maxWeaken')}")
    print(f"homeFreeRam:      {status.get('homeFreeRam', 0):.2f} GB")
    print(f"hacked:           {fmt_num(status.get('hacked', 0))}")
    print(f"rate:             {fmt_num(status.get('rate', 0))}/s")
    print(f"avgRate:          {fmt_num(status.get('avgRate', 0))}/s")
    print(f"totalHacked:      {fmt_num(status.get('totalHacked', 0))}")
    print(f"workers:          {len(status.get('workers', []))}")
    print(f"candidate:        {status.get('candidate')}")
    print(f"candidateScore:   {fmt_num(status.get('candidateScore', 0))}")
    print(f"expectedIncome:   {fmt_num(status.get('candidateExpectedIncome', 0))}/s")
    print('\nworkers:')
    for worker in status.get('workers', []):
        actions = worker.get('actions') or []
        action_str = ', '.join(f"{a['script']}({a['threads']})" for a in actions) or 'idle'
        print(f"  - {worker['host']}: freeRam={worker.get('freeRam',0):.2f} GB usedRam={worker.get('usedRam',0):.2f} GB maxRam={worker.get('maxRam',0):.2f} GB actions={action_str}")


if __name__ == '__main__':
    main()
